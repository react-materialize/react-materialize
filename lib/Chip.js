'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

var Chip = function Chip(_ref) {
  var children = _ref.children,
      close = _ref.close,
      className = _ref.className,
      props = _objectWithoutProperties(_ref, ['children', 'close', 'className']);

  return _react2.default.createElement(
    'div',
    _extends({ className: (0, _classnames2.default)('chip', className) }, props),
    children,
    close ? _react2.default.createElement(
      'i',
      { className: 'close material-icons' },
      'close'
    ) : null
  );
};

Chip.propTypes = {
  className: _propTypes2.default.string,
  children: _propTypes2.default.node,
  /**
   * Shows a close icon
   */
  close: _propTypes2.default.bool
};

Chip.defaultProps = {
  close: false
};

exports.default = Chip;