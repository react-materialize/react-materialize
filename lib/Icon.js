'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _constants = require('./constants');

var _constants2 = _interopRequireDefault(_constants);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var Icon = function (_React$Component) {
  _inherits(Icon, _React$Component);

  function Icon() {
    _classCallCheck(this, Icon);

    return _possibleConstructorReturn(this, (Icon.__proto__ || Object.getPrototypeOf(Icon)).apply(this, arguments));
  }

  _createClass(Icon, [{
    key: 'render',
    value: function render() {
      var _this2 = this;

      var classes = {
        'material-icons': true
      };
      _constants2.default.PLACEMENTS.forEach(function (p) {
        classes[p] = _this2.props[p];
      });
      _constants2.default.ICON_SIZES.forEach(function (s) {
        classes[s] = _this2.props[s];
      });
      return _react2.default.createElement(
        'i',
        { className: (0, _classnames2.default)(classes, this.props.className) },
        this.props.children
      );
    }
  }]);

  return Icon;
}(_react2.default.Component);

Icon.propTypes = {
  /*
   * Classname passed to i tag
   */
  className: _propTypes2.default.string,
  /*
   * Icon type: <a href='https://material.io/icons/'>https://material.io/icons/</a>
   */
  children: _propTypes2.default.string,
  /*
   * Placement for icon if used beside a text ↓
   */
  left: _propTypes2.default.bool,
  center: _propTypes2.default.bool,
  right: _propTypes2.default.bool,
  /*
   * Sizes for icons ↓
   */
  tiny: _propTypes2.default.bool,
  small: _propTypes2.default.bool,
  medium: _propTypes2.default.bool,
  large: _propTypes2.default.bool
};

exports.default = Icon;