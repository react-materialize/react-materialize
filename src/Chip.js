import React from 'react';
import PropTypes from 'prop-types';

const Chip = ({ children, close }) => {
  return (
    <div className='chip'>
      {children}
      {close ? <i className='close material-icons'>close</i> : null}
    </div>
  );
};

Chip.propTypes = {
  children: PropTypes.node,
  /**
   * If show a close icon
   * @default false
   */
  close: PropTypes.bool
};

export default Chip;
