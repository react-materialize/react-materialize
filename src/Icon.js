import React from 'react';
import PropTypes from 'prop-types';
import constants from './constants';
import cx from 'classnames';

class Icon extends React.Component {
  render() {
    const classes = {
      'material-icons': true
    };
    constants.PLACEMENTS.forEach(p => {
      classes[p] = this.props[p];
    });
    constants.ICON_SIZES.forEach(s => {
      classes[s] = this.props[s];
    });
    return (
      <i className={cx(classes, this.props.className)}>{this.props.children}</i>
    );
  }
}

Icon.propTypes = {
  /*
   * Classname passed to i tag
   */
  className: PropTypes.string,
  /*
   * Icon type: <a href='https://material.io/icons/'>https://material.io/icons/</a>
   */
  children: PropTypes.string,
  /*
   * Placement for icon if used beside a text ↓
   */
  left: PropTypes.bool,
  center: PropTypes.bool,
  right: PropTypes.bool,
  /*
   * Sizes for icons ↓
   */
  tiny: PropTypes.bool,
  small: PropTypes.bool,
  medium: PropTypes.bool,
  large: PropTypes.bool
};

export default Icon;
