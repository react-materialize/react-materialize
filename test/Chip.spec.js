import React from 'react';
import { shallow, mount } from 'enzyme';
import Chip from '../src/Chip';
import Icon from '../src/Icon';
import mocker from './helper/new-mocker';

describe('<Chip />', () => {
  let wrapper;

  test('should render', () => {
    wrapper = shallow(<Chip />);
    expect(wrapper).toMatchSnapshot();
  });

  test('accepts a close prop', () => {
    wrapper = shallow(<Chip close />);
    expect(wrapper.find(Icon)).toHaveLength(1);
    expect(wrapper.find(Icon).hasClass('close')).toEqual(true);
  });

  test('accepts children', () => {
    let child = <p className="child">Child</p>;
    wrapper = shallow(<Chip>{child}</Chip>);
    expect(wrapper).toMatchSnapshot();
  });

  test('accepts className', () => {
    wrapper = shallow(<Chip className="find" />);
    expect(wrapper.hasClass('find')).toBeTruthy();
  });

  describe('initialises', () => {
    const chipInitMock = jest.fn();
    const chipInstanceDestroyMock = jest.fn();
    const chipMock = {
      init: (el, options) => {
        chipInitMock(options);
        return {
          destroy: chipInstanceDestroyMock
        };
      }
    };
    const restore = mocker('Chips', chipMock);

    beforeEach(() => {
      chipInitMock.mockClear();
      chipInstanceDestroyMock.mockClear();
    });

    afterAll(() => {
      restore();
    });

    test('calls Chips', () => {
      const options = {
        data: [
          {
            tag: 'Apple'
          },
          {
            tag: 'Microsoft'
          },
          {
            tag: 'Google'
          }
        ],
        placeholder: 'Enter a tag',
        secondaryPlaceholder: '+Tag',
        autocompleteOptions: {
          data: {
            Apple: null,
            Microsoft: null,
            Google: null
          },
          limit: Infinity
        }
      };

      mount(<Chip options={options} />);

      expect(chipInitMock).toHaveBeenCalledWith(options);
    });
  });
});
