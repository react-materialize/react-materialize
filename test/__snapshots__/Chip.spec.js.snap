// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Chip /> accepts children 1`] = `
<div
  className="chip"
>
  <p
    className="child"
  >
    Child
  </p>
</div>
`;

exports[`<Chip /> should render 1`] = `
<div
  className="chip"
/>
`;
